pkg/ai/openai.go

```diff
@@ -75,25 +75,21 @@ func (a *OpenAIClient) Parse(ctx context.Context, prompt []string, cache cache.I
 	// Check for cached data
 	sEnc := base64.StdEncoding.EncodeToString([]byte(inputKey))
 	cacheKey := util.GetCacheKey(a.GetName(), a.language, sEnc)
-	// find in viper cache
-	if cache.Exists(cacheKey) {
-		// retrieve data from cache
-		response, err := cache.Load(cacheKey)
 
+	if !cache.IsCacheDisabled() {
```

Comment:
This is a bit of a leaky abstraction imho and complicates the code. By having the `noop` cache
implementation, there was no need to test if caching is disabled or not. Now you have to add an 
additional `if` statement, everytime you want to use the cache.